options.go

```diff
@@ -38,6 +41,13 @@ func WithDefaults() HealthOption {
 	return healthOptionFunc(func(h *health) {})
 }
 
+// WithContext sets the main context.Context of the Health object. By default, context.Background is used.
+func WithContext(ctx context.Context) HealthOption {
```

Comment:
I find this slightly problematic: let's say you pass a context to the health checker with a short deadline (by accident), but no timeout. This would result in health checks failing after a certain period of time.

It's a hard to find bug (`context deadline exceeded` is not a very descriptive error and a bad context wouldn't be my first thought) and can easily bring down your service.

I would either add a warning here with the above risks, or check if there is a deadline set in the context and reject it if there is one, log a warning, etc. I'm not saying I know the perfect solution, but this seemingly simple function can cause a lot of harm.

Reply:
The difference is that context is usually passed to individual requests (in which case deadline makes perfect sense), whereas in this case you pass in the context for every request, so a global deadline simply doesn't make sense.

Reply:
I agree. I don't think this is useful for the users, while it's bug prone, and may get confusing